File: OrchidCore/src/main/java/com/eden/orchid/api/options/archetypes/EnvironmentVariableArchetype.java
package com.eden.orchid.api.options.archetypes;

import com.eden.orchid.api.options.OptionArchetype;
import com.eden.orchid.api.options.annotations.Description;

import java.util.Map;

@Description(value = "Loads additional values from secure environment variables.",
        name = "Environment Variables"
)
public class EnvironmentVariableArchetype implements OptionArchetype {

    @Override
    public Map<String, Object> getOptions(Object target, String archetypeKey) {
        Map<String, ?> vars = System.getenv();
        return (Map<String, Object>) vars;
    }

}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/archetypes/EnvironmentVariableArchetype.kt
package com.eden.orchid.api.options.archetypes

import com.eden.orchid.api.options.OptionArchetype
import com.eden.orchid.api.options.annotations.Description

@Description(value = "Loads additional values from secure environment variables.", name = "Environment Variables")
class EnvironmentVariableArchetype : OptionArchetype {

    override fun getOptions(target: Any, archetypeKey: String): Map<String, Any> {
        val vars = System.getenv()
        return vars as Map<String, Any>
    }

}
